### PR TITLE
[netcore] Fix DynamicMethodToString test

### PR DESCRIPTION
**Pull request by EgorBo**
Fixes https://github.com/mono/mono/issues/15323

### CODE REVIEW

File: netcore/System.Private.CoreLib/src/System.Reflection.Emit/DynamicMethod.cs [hunk 1]
```diff
@@ -316,14 +317,25 @@ internal override int GetParametersCount ()
...
 			}
-			return ReturnType.Name+" "+Name+"("+parms+")";
+
+			string returnTypeName;
+			if (ReturnType == typeof (void))
+				returnTypeName = "Void";
+			else if (ReturnType.IsPrimitive)
+				returnTypeName = ReturnType.Name;
+			else
+				returnTypeName = ReturnType.ToString ();
```
Comment (filipnavara) [inline netcore/System.Private.CoreLib/src/System.Reflection.Emit/DynamicMethod.cs]: CoreCLR also has exception for `TypedReference` (https://github.com/dotnet/coreclr/blob/2ec0bcfefe0bccb9a7023b8d528c775349d93458/src/System.Private.CoreLib/src/System/RuntimeType.CoreCLR.cs#L3904). Do we want to special case that one too?

File: netcore/System.Private.CoreLib/src/System.Reflection.Emit/DynamicMethod.cs [hunk 2]
```diff
@@ -316,14 +317,25 @@ internal override int GetParametersCount ()
...
+			var paramsSb = new StringBuilder ();
+			ParameterInfo [] parameters = GetParametersInternal ();
+			for (int i = 0; i < parameters.Length; ++i) {
+				ParameterInfo p = parameters [i];
 				if (i > 0)
-					parms = parms + ", ";
-				parms = parms + p [i].ParameterType.Name;
+					paramsSb.Append (", ");
+
+				paramsSb.Append (p.ParameterType.IsPrimitive ? p.ParameterType.Name : p.ParameterType.ToString ());
```
Comment (filipnavara) [inline netcore/System.Private.CoreLib/src/System.Reflection.Emit/DynamicMethod.cs]: Same nit about `TypedReference` as below. ByRef handling is also missing (https://github.com/dotnet/coreclr/blob/9773db1e7b1acb3ec75c9cc0e36bd62dcbacd6d5/src/System.Private.CoreLib/src/System/Reflection/MethodBase.CoreCLR.cs#L73-L77) but the usability of that code is questionable.

File: netcore/System.Private.CoreLib/src/System.Reflection.Emit/DynamicMethod.cs (hunk 1 above)
Comment (EgorBo) [reply netcore/System.Private.CoreLib/src/System.Reflection.Emit/DynamicMethod.cs]: Good catch!

File: netcore/CoreFX.issues.rsp [hunk 3]
```diff
@@ -952,15 +952,6 @@
 # https://github.com/mono/mono/issues/15192
 -nomethod System.Reflection.Tests.AssemblyTests.LoadFromStream_Location_IsEmpty
 
-####################################################################
-##  System.Reflection.Metadata.Tests
-####################################################################
-
-# Expected null, got a double byte array
-# https://github.com/mono/mono/issues/15193
```
Comment (vargaz) [inline netcore/CoreFX.issues.rsp]: Unrelated change ?

Comment (EgorBo) [reply netcore/CoreFX.issues.rsp]: yeah, I'll revise all `-nomethod` in a separate PR

**Comment by EgorBo** [comment]
Closing it - going to move needed pieces from CoreCLR to shared and re-use them after.
